dbrequests/mysql/connection.py

```diff
@@ -44,9 +44,15 @@ def _send_data_update(self, df, table, mode='replace', **params):
             self._insert_update(df, table, tmp_table)
 
     def _write_csv(self, df, file):
-        # We have to convert any obj64 types to str64: Frame.to_csv can't
-        # handle obj64 types on it's own.
-        df = df[:, f[:].remove(f[obj64]).extend(str64(f[obj64]))][:, df.names]
+        # Before writing, we need to convert all columns to strings for two
+        # reasons:
+        # - We have to convert any obj64 types to str64: Frame.to_csv can't
+        #   process them.
+        # - We have to replace None with NULL to tell MySQL, that we have
+        #   actual NULL values. An empty cell is sometimes, but not always a
+        #   NULL value. See #30
+        df = df[:, f[:].remove(f[:]).extend(str64(f[:]))][:, df.names]
```

Comment:
`f` is a great function name xD
I do not understand what is happening here specifically except that str64 is happenning

There is quite a lot happening in this one line. I count 8? things happening. Maybe separate it into multiple lines. 
Though that is not as concise, it is better to maintain in the long run because youll understand fast coming back in two month, why f.remove(f) is happening.

Reply:
I believe this is idiomatic datatable code. See the docs, I copied it from there: https://datatable.readthedocs.io/en/v0.10.1/f-expressions.html#modifying-a-columnset. At most, we could split it into two lines, and even that would not help readability. The tricky part of this expression is 

```
f[:].remove(f[:]).extend(str64(f[:]))
```

and it means: from the complete DT (f[:]) remove all columns (remove(f[:])) and extend it by all columns coerced into strings (extend(str64(f[:]))). Or in other words, convert all columns to strings. But in datatable, you cannot replace a column, with a different type. This is due to some performance optimization. It means we first have to add a new column and then delete the old ones. This happens in one expression, to take advantage of internal optimizations, I believe.

Reply:
alright